src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CollectionExpression.cs

```diff
@@ -98,16 +98,25 @@ private BoundExpression VisitArrayOrSpanCollectionExpression(BoundCollectionExpr
                     collectionTypeKind == CollectionExpressionTypeKind.Span ? WellKnownType.System_Span_T : WellKnownType.System_ReadOnlySpan_T), TypeCompareKind.AllIgnoreOptions));
                 Debug.Assert(elementType.Equals(spanType.TypeArgumentsWithAnnotationsNoUseSiteDiagnostics[0], TypeCompareKind.AllIgnoreOptions));
 
+                if (elements.Length == 0)
```

Comment:
I felt it was worth basing this PR on #70260 because the presence of the optimization affects whether we want to give this warning.